### DISCUSSION TITLE
Trying to display one to many relations in view

### DISCUSSION THREAD

**Discussion by sum117** [pocketbase/pocketbase#2308] [Q&A] [answered]
I'm currently getting it done using the query:

```sql
SELECT
  r.id,
  r.name,
  JSON_GROUP_ARRAY(c.id) as characters, 
  COUNT(c.id) AS population
FROM 
   races r
  INNER JOIN characters c ON c.race = r.id
  
GROUP BY
  r.id;
```

But this doesn't look very good:

![Screenshot_2023-04-17-02-33-45-816_com.android.chrome-edit.jpg](https://user-images.githubusercontent.com/75037449/232390581-b108c641-d4c8-4e9c-bda3-3da97277ac89.jpg)

Is there any way to achieve this?

![Screenshot_2023-04-17-02-34-31-121_com.android.chrome-edit.jpg](https://user-images.githubusercontent.com/75037449/232390905-177fb1a3-bf2c-4540-94d6-f2505060431b.jpg)

I tried to remove the JSON_GROUP_ARRAY, but then it only displays one of the characters.

Thanks for your time 🙏

**Comment by ganigeorgiev** [comment]
SQL expressions (in your case `JSON_GROUP_ARRAY`) are resolved as `json` type field.

Only single `id` columns and simple `relation` fields references are resolved as a `relation` field in a View collection schema.

**Reply by sum117** [reply]
I see. I'm trying to do this because I'm having issues with syncing relations. I made a one to many relation between races and characters, and then character->race for the characters collection.

Issue is that everytime I update the character on that spot I also have to update the race, making me use two queries. Do you have any suggestions?

**Reply by ganigeorgiev** [reply]
> I made a one to many relation between races and characters, and then character->race for the characters collection.

But why? Please elaborate, I have no context of what `race` and `character` represent.

**Reply by sum117** [reply]
Sorry.

This is what I meant:
![Screenshot_2023-04-17-02-53-56-259_com.android.chrome-edit.jpg](https://user-images.githubusercontent.com/75037449/232394858-f38a2994-16e5-44c0-bf95-e367b9736a47.jpg)


![Screenshot_2023-04-17-02-53-35-570_com.android.chrome-edit.jpg](https://user-images.githubusercontent.com/75037449/232394872-df4d2f8e-ed01-4c6b-8798-e3738f3c7f33.jpg)


I'm making a text based game and I made a characters relation in each race so I can query them easily, but I also linked the race in the character themselves. I just want to know if there's a better way of doing this. 

I hope this is enough context, I'm not very acquainted with this scenario.

My issue is in syncing them. Once I change a character's race, I have to do another query on the race to remove it from there too, since the relations aren't interacting.

**Reply by sum117** [reply]
```ts
  private static async updateEntityWithCharacter<T extends RelationFields>( 
     entity: T, 
     character: Character 
   ): Promise<void> { 
     const entityType = entity.collectionName as keyof typeof COLLECTIONS; 
  
     if ("character" in entity) { 
       entity.character = character.id; 
     } else if ("characters" in entity && Array.isArray(entity.characters)) { 
       entity.characters = [...entity.characters, character.id]; 
     } 
  
     await PocketBase.updateEntity<T>({ 
       entityType, 
       entityData: entity, 
     }); 
   } 
 }
```

**Reply by ganigeorgiev** [reply, marked as answer]
I'm sorry, I still don't understand what is the supposed relation between characters and races, what actually they represent and what you are trying to do in general.

But in any case, you don't have to create the relation in 2 places. Instead define it only in one of the collection and if you want to expand the related record with the main one, you can check [Expanding relations](https://pocketbase.io/docs/expanding-relations/) (for both direct and indirect relations).

**Reply by sum117** [reply]
Alright. I'm sorry for this. I'm not very social! I will read further. Thank you so much for helping me 🙏

**Reply by sum117** [reply]
Hey, just wanted to thank you. Turns out indirect relationships was exactly what I needed! Very powerful indeed, TYSM!
